File: 0080/solution.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int write_to = 0;
        int count = 0;
        int prev = (int) Math.pow(10, 5);
        for (int num : nums) {
            if (num != prev) {
                count = 1;
            } else {
                count++;
            }
            if (count <= 2) {
                nums[write_to++] = num;
            }
            prev = num;
        }
        return write_to;
    }
}

File: 0080/solution.py
class Solution:
    def removeDuplicates(self, nums: list[int]) -> int:
        write_to = 0
        last_number = float("inf")
        count_last = 1
        k = 0
        for i in range(len(nums)):
            if nums[i] != last_number:
                nums[write_to] = nums[i]
                last_number = nums[i]
                write_to += 1
                count_last = 1
                k += 1
            elif count_last < 2:
                nums[write_to] = nums[i]
                write_to += 1
                count_last += 1
                k += 1
        return k

        # time O(n)
        # space O(1)
